src/clients/python/src/tigerbeetle_python/bindings.py

```diff
@@ -0,0 +1,733 @@
+##########################################################
+## This file was auto-generated by tb_client_header.zig ##
+##              Do not manually modify.                 ##
+##########################################################
+from __future__ import annotations
+
+import ctypes
+import dataclasses
+import enum
+from collections.abc import Callable # noqa: TCH003
+from typing import Any
+
+from .lib import c_uint128, tbclient, IntegerOverflowError
+
+
+class Operation(enum.IntEnum):
+    PULSE = 128
+    CREATE_ACCOUNTS = 129
+    CREATE_TRANSFERS = 130
+    LOOKUP_ACCOUNTS = 131
+    LOOKUP_TRANSFERS = 132
+    GET_ACCOUNT_TRANSFERS = 133
+    GET_ACCOUNT_BALANCES = 134
+    QUERY_ACCOUNTS = 135
+    QUERY_TRANSFERS = 136
+
+
+class PacketStatus(enum.IntEnum):
+    OK = 0
+    TOO_MUCH_DATA = 1
+    CLIENT_EVICTED = 2
+    CLIENT_SHUTDOWN = 3
+    INVALID_OPERATION = 4
+    INVALID_DATA_SIZE = 5
+
+
+class CPacket(ctypes.Structure):
+    @classmethod
+    def from_param(cls, obj):
+        if obj.operation > 2**8 - 1:
```

Comment:
Do we need to check that it is non-negative as well? If we do, then I think we could use a helper function for validation.

Reply:
Ahhh, let's maybe do 

```
operation = ctypes.c_int8(obj.operation)
```

and rely on the build-in range checks?

I don't know what's the right way to handle duck typing here, but this way we offload that question to Python's stdlib!

Reply:
> Do we need to check that it is non-negative as well

Yes, good catch!

> and rely on the build-in range checks?

```
>>> ctypes.c_uint8(-10000000)
c_ubyte(128)
```

These are not the [built-in range checks](https://docs.python.org/3/library/ctypes.html#ctypes.c_byte) you're looking for.